### PR TITLE
wOAMBuffer -> wShadowOAM

### PR DESCRIPTION
**Pull request by coltongit**
ISSOtm, as he pointed out in https://github.com/pret/pokecrystal/pull/943#discussion_r906649489, noticed that the WRAM constant `wOAMBuffer` is, by most usage and convention elsewhere, known as `wShadowOAM` instead.
This PR aims to rename this particular WRAM constant so that it matches this convention.
pokegold companion PR: https://github.com/pret/pokegold/pull/99
pokegold-spaceworld companion PR: https://github.com/pret/pokegold-spaceworld/pull/103
pokecrystal companion PR: https://github.com/pret/pokecrystal/pull/971
This also works toward bullet point 3 in #315 by syncing this OAM constant across all Gen 1 and 2 main series pret repos

### CODE REVIEW

File: engine/battle/animations.asm [hunk 1]
```diff
@@ -147,7 +147,7 @@ DrawFrameBlock:
 	jr z, .resetFrameBlockDestAddr
 	call AnimationCleanOAM
 .resetFrameBlockDestAddr
-	ld hl, wOAMBuffer ; OAM buffer
+	ld hl, wShadowOAM ; OAM buffer
```
Comment (Rangi42) [inline engine/battle/animations.asm:150]:
```suggestion
	ld hl, wShadowOAM
```

File: engine/battle/animations.asm [hunk 2]
```diff
@@ -583,7 +583,7 @@ PlaySubanimation:
 	call GetMoveSound
 	call PlaySound
 .skipPlayingSound
-	ld hl, wOAMBuffer ; base address of OAM buffer
+	ld hl, wShadowOAM ; base address of OAM buffer
```
Comment (Rangi42) [inline engine/battle/animations.asm:586]:
```suggestion
	ld hl, wShadowOAM
```

File: engine/battle/animations.asm [hunk 3]
```diff
@@ -855,7 +855,7 @@ TradeShakePokeball:
 ; if it's the end of the animation, make the ball jump up
 	ld de, BallMoveDistances1
 .loop
-	ld hl, wOAMBuffer ; OAM buffer
+	ld hl, wShadowOAM ; OAM buffer
```
Comment (Rangi42) [inline engine/battle/animations.asm:858]:
```suggestion
	ld hl, wShadowOAM
```

File: engine/battle/animations.asm [hunk 4]
```diff
@@ -885,7 +885,7 @@ BallMoveDistances1:
 TradeJumpPokeball:
 	ld de, BallMoveDistances2
 .loop
-	ld hl, wOAMBuffer ; OAM buffer
+	ld hl, wShadowOAM ; OAM buffer
```
Comment (Rangi42) [inline engine/battle/animations.asm:888]:
```suggestion
	ld hl, wShadowOAM
```

File: engine/battle/animations.asm [hunk 5]
```diff
@@ -925,8 +925,8 @@ BallMoveDistances2:
 ; this function copies the current musical note graphic
 ; so that there are two musical notes flying towards the defending pokemon
 DoGrowlSpecialEffects:
-	ld hl, wOAMBuffer ; OAM buffer
-	ld de, wOAMBuffer + $10
+	ld hl, wShadowOAM ; OAM buffer
```
Comment (Rangi42) [inline engine/battle/animations.asm:928]:
```suggestion
	ld hl, wShadowOAM
```

File: engine/battle/animations.asm [hunk 6]
```diff
@@ -925,8 +925,8 @@ BallMoveDistances2:
...
+	ld de, wShadowOAM + $10
```
Comment (Rangi42) [inline engine/battle/animations.asm:929]:
```suggestion
	ld de, wShadowOAMSprite04
```

File: engine/battle/animations.asm [hunk 7]
```diff
@@ -2433,7 +2433,7 @@ FallingObjects_UpdateMovementByte:
 	ret
 
 FallingObjects_InitXCoords:
-	ld hl, wOAMBuffer + $01
+	ld hl, wShadowOAM + $01
```
Comment (Rangi42) [inline engine/battle/animations.asm:2436]:
```suggestion
	ld hl, wShadowOAMSprite00XCoord
```

File: engine/battle/battle_transitions.asm [hunk 8]
```diff
@@ -26,7 +26,7 @@ BattleTransition:
 	jr nz, .loop1
 
 ; Clear OAM except for the blocks used by the player and enemy trainer sprites.
-	ld hl, wOAMBuffer + $10
+	ld hl, wShadowOAM + $10
```
Comment (Rangi42) [inline engine/battle/battle_transitions.asm:29]:
```suggestion
	ld hl, wShadowOAMSprite04
```

File: engine/battle/core.asm [hunk 9]
```diff
@@ -105,7 +105,7 @@ SlidePlayerAndEnemySilhouettesOnScreen:
 ; instead, the enemy pic is part of the background and uses the scroll register, while the player's head is a sprite and is slid by changing its X coordinates in a loop
 SlidePlayerHeadLeft:
 	push bc
-	ld hl, wOAMBuffer + $01
+	ld hl, wShadowOAM + $01
```
Comment (Rangi42) [inline engine/battle/core.asm:108]:
```suggestion
	ld hl, wShadowOAMSprite00XCoord
```

File: engine/battle/draw_hud_pokeball_gfx.asm [hunk 10]
```diff
@@ -183,7 +183,7 @@ SetupPlayerAndEnemyPokeballs:
 	ld a, $50
 	ld [hli], a
 	ld [hl], $68
-	ld hl, wOAMBuffer + $18
+	ld hl, wShadowOAM + $18
```
Comment (Rangi42) [inline engine/battle/draw_hud_pokeball_gfx.asm:186]:
```suggestion
	ld hl, wShadowOAMSprite06
```

File: engine/events/diploma.asm [hunk 11]
```diff
@@ -42,7 +42,7 @@ DisplayDiploma::
 
 ; Move the player 33 pixels right and set the priority bit so he appears
 ; behind the background layer.
-	ld hl, wOAMBuffer + $01
+	ld hl, wShadowOAM + $01
```
Comment (Rangi42) [inline engine/events/diploma.asm:45]:
```suggestion
	ld hl, wShadowOAMSprite00XCoord
```

File: engine/gfx/mon_icons.asm [hunk 12]
```diff
@@ -43,15 +43,15 @@ GetAnimationSpeed:
...
+	ld de, wShadowOAM
 	ld bc, $60
 	call CopyData
 	pop bc
 	xor a
 	jr .incTimer
 .animateSprite
 	push bc
-	ld hl, wOAMBuffer + $02 ; OAM tile id
+	ld hl, wShadowOAM + $02 ; OAM tile id
```
Comment (Rangi42) [inline engine/gfx/mon_icons.asm:54]:
```suggestion
	ld hl, wShadowOAMSprite00TileID
```

File: engine/items/item_effects.asm [hunk 13]
```diff
@@ -1175,7 +1175,7 @@ ItemUseMedicine:
 	xor a
 	ld [wBattleMonStatus], a ; remove the status ailment in the in-battle pokemon data
 .calculateHPBarCoords
-	ld hl, wOAMBuffer + $90
+	ld hl, wShadowOAM + $90
```
Comment (Rangi42) [inline engine/items/item_effects.asm:1178]:
```suggestion
	ld hl, wShadowOAMSprite36
```

File: engine/items/town_map.asm [hunk 14]
```diff
@@ -47,7 +47,7 @@ DisplayTownMap:
 	call TownMapCoordsToOAMCoords
 	ld a, $4
 	ld [wOAMBaseTile], a
-	ld hl, wOAMBuffer + $10
+	ld hl, wShadowOAM + $10
```
Comment (Rangi42) [inline engine/items/town_map.asm:50]:
```suggestion
	ld hl, wShadowOAMSprite04
```

File: engine/items/town_map.asm [hunk 15]
```diff
@@ -60,7 +60,7 @@ DisplayTownMap:
 	hlcoord 1, 0
 	ld de, wcd6d
 	call PlaceString
-	ld hl, wOAMBuffer + $10
+	ld hl, wShadowOAM + $10
```
Comment (Rangi42) [inline engine/items/town_map.asm:63]:
```suggestion
	ld hl, wShadowOAMSprite04
```

File: engine/items/town_map.asm [hunk 16]
```diff
@@ -439,9 +439,9 @@ TownMapCoordsToOAMCoords:
 WritePlayerOrBirdSpriteOAM:
 	ld a, [wOAMBaseTile]
 	and a
-	ld hl, wOAMBuffer + $90 ; for player sprite
+	ld hl, wShadowOAM + $90 ; for player sprite
```
Comment (Rangi42) [inline engine/items/town_map.asm:442]:
```suggestion
	ld hl, wShadowOAMSprite36 ; for player sprite
```

File: engine/items/town_map.asm [hunk 17]
```diff
@@ -439,9 +439,9 @@ TownMapCoordsToOAMCoords:
...
 	jr z, WriteTownMapSpriteOAM
-	ld hl, wOAMBuffer + $80 ; for bird sprite
+	ld hl, wShadowOAM + $80 ; for bird sprite
```
Comment (Rangi42) [inline engine/items/town_map.asm:444]:
```suggestion
	ld hl, wShadowOAMSprite32 ; for bird sprite
```

File: engine/movie/splash.asm [hunk 18]
```diff
@@ -16,11 +16,11 @@ LoadShootingStarGraphics:
 	lb bc, BANK(FallingStar), (FallingStarEnd - FallingStar) / $10
 	call CopyVideoData
 	ld hl, GameFreakLogoOAMData
-	ld de, wOAMBuffer + $60
+	ld de, wShadowOAM + $60
```
Comment (Rangi42) [inline engine/movie/splash.asm:19]: ...you get the idea: use the labels generated by the `sprite_oam_struct` macro, and remove redundant comments.

**Review by vulcandth — APPROVED** [review]
There you go. This looks good; thanks.

File: engine/movie/splash.asm [hunk 19]
```diff
@@ -16,11 +16,11 @@ LoadShootingStarGraphics:
 	lb bc, BANK(FallingStar), (FallingStarEnd - FallingStar) / $10
 	call CopyVideoData
 	ld hl, GameFreakLogoOAMData
-	ld de, wOAMBuffer + $60
+	ld de, wShadowOAMSprite24
 	ld bc, GameFreakLogoOAMDataEnd - GameFreakLogoOAMData
 	call CopyData
 	ld hl, GameFreakShootingStarOAMData
-	ld de, wOAMBuffer
+	ld de, wShadowOAM
```
Comment (Rangi42) [inline engine/movie/splash.asm:23]:
```suggestion
	ld de, wShadowOAMSprite00
```

File: engine/movie/splash.asm [hunk 20]
```diff
@@ -30,7 +30,7 @@ AnimateShootingStar:
 	call PlaySound
 
 ; Move the big star down and left across the screen.
-	ld hl, wOAMBuffer
+	ld hl, wShadowOAM
```
Comment (Rangi42) [inline engine/movie/splash.asm:33]:
```suggestion
	ld hl, wShadowOAMSprite00
```

File: engine/movie/title2.asm [hunk 21]
```diff
@@ -115,6 +115,6 @@ GetTitleBallY:
 	pop de
 	and a
 	ret z
-	ld [wOAMBuffer + $28], a
+	ld [wShadowOAMSprite10], a
```
Comment (Rangi42) [inline engine/movie/title2.asm:118]:
```suggestion
	ld [wShadowOAMSprite10YCoord], a
```

File: engine/overworld/cut2.asm [hunk 22]
```diff
@@ -33,7 +33,7 @@ AnimCut:
 	ld c, $8
 	call AnimCutGrass_UpdateOAMEntries
 	call AnimCutGrass_SwapOAMEntries
-	ld hl, wOAMBuffer + $90
+	ld hl, wShadowOAMSprite36
```
Comment (Rangi42) [inline engine/overworld/cut2.asm:36]:
```suggestion
	ld hl, wShadowOAMSprite36YCoord
```

File: engine/overworld/player_animations.asm [hunk 23]
```diff
@@ -205,19 +205,19 @@ LeaveMapThroughHoleAnim:
...
-	ld [wOAMBuffer + 3 * 4 + 2], a
+	ld a, [wShadowOAMSprite00TileID]
+	ld [wShadowOAMSprite02TileID], a
+	ld a, [wShadowOAMSprite01TileID]
+	ld [wShadowOAMSprite03TileID], a
 	ld a, $a0
-	ld [wOAMBuffer + 0 * 4], a
-	ld [wOAMBuffer + 1 * 4], a
+	ld [wShadowOAMSprite00], a
+	ld [wShadowOAMSprite01], a
```
Comment (Rangi42) [inline engine/overworld/player_animations.asm:214]:
```suggestion
	ld [wShadowOAMSprite00TileID], a
	ld [wShadowOAMSprite01TileID], a
```

File: engine/overworld/player_animations.asm [hunk 24]
```diff
@@ -205,19 +205,19 @@ LeaveMapThroughHoleAnim:
...
 	ld c, 2
 	call DelayFrames
 	; hide upper half of player's sprite
 	ld a, $a0
-	ld [wOAMBuffer + 2 * 4], a
-	ld [wOAMBuffer + 3 * 4], a
+	ld [wShadowOAMSprite02], a
+	ld [wShadowOAMSprite03], a
```
Comment (Rangi42) [inline engine/overworld/player_animations.asm:220]:
```suggestion
	ld [wShadowOAMSprite02TileID], a
	ld [wShadowOAMSprite03TileID], a
```

File: engine/overworld/player_animations.asm [hunk 25]
```diff
@@ -424,7 +424,7 @@ FishingAnim:
 	cp SPRITE_FACING_UP
 	jr nz, .skipHidingFishingRod
 	ld a, $a0
-	ld [wOAMBuffer + $9c], a
+	ld [wShadowOAMSprite39], a
```
Comment (Rangi42) [inline engine/overworld/player_animations.asm:427]:
```suggestion
	ld [wShadowOAMSprite39TileID], a
```

File: engine/overworld/player_animations.asm [hunk 26]
```diff
@@ -438,7 +438,7 @@ FishingAnim:
 	cp SPRITE_FACING_UP
 	jr nz, .skipUnhidingFishingRod
 	ld a, $44
-	ld [wOAMBuffer + $9c], a
+	ld [wShadowOAMSprite39], a
```
Comment (Rangi42) [inline engine/overworld/player_animations.asm:441]:
```suggestion
	ld [wShadowOAMSprite39TileID], a
```

File: engine/slots/slot_machine.asm [hunk 27]
```diff
@@ -757,23 +757,23 @@ SlotMachine_UpdateBallTiles:
 SlotMachine_AnimWheel1:
 	ld bc, SlotMachineWheel1
 	ld de, wSlotMachineWheel1Offset
-	ld hl, wOAMBuffer
+	ld hl, wShadowOAM
```
Comment (Rangi42) [inline engine/slots/slot_machine.asm:760]:
```suggestion
	ld hl, wShadowOAMSprite00
```

File: engine/overworld/player_animations.asm (hunk 25 above)
Comment (vulcandth) [reply engine/overworld/player_animations.asm:427]:
```suggestion
	ld [wShadowOAMSprite39YCoord], a
```

File: engine/overworld/player_animations.asm (hunk 26 above)
Comment (vulcandth) [reply engine/overworld/player_animations.asm:441]:
```suggestion
	ld [wShadowOAMSprite39YCoord], a
```

**Review by Rangi42 — APPROVED** [review]

**Pull request merged**